src/audio-worklet/free-queue/examples/simple-passthrough/worker.js

```diff
@@ -0,0 +1,38 @@
+import FreeQueue from "../../src/free-queue.js";
+import { FRAME_SIZE } from "./constants.js";
+
+let queue;
+let atomicState;
+
+/**
+ * Worker message event handler.
+ * This will initialize worker with FreeQueue instance and set loop for audio
+ * processing. 
+ */
+self.onmessage = (msg) => {
+  if (msg.data.type === "init") {
+    let { inputQueue, outputQueue, atomicState } = msg.data.data;
+    Object.setPrototypeOf(inputQueue, FreeQueue.prototype);
+    Object.setPrototypeOf(outputQueue, FreeQueue.prototype);
+    
+    // buffer for storing data pulled out from queue.
+    const input = new Float32Array(FRAME_SIZE);
+    // loop for processing data.
+    while (Atomics.wait(atomicState, 0, 0) === 'ok') {
```

Comment:
I am curious how this shows up on the JS profilier. Is this a busy loop? Or does it automatically yield?

Reply:
I am not sure how to use JS Profiler correctly in this case, and extract information from it